lib/vssh/libssh2.c

```diff
@@ -435,22 +435,68 @@ static int sshkeycallback(struct Curl_easy *easy,
 #ifdef HAVE_LIBSSH2_SESSION_HANDSHAKE
 #define libssh2_session_startup(x,y) libssh2_session_handshake(x,y)
 #endif
+static int convert_ssh2_keytype(int sshkeytype)
+{
+  int keytype = CURLKHTYPE_UNKNOWN;
+  switch(sshkeytype) {
+  case LIBSSH2_HOSTKEY_TYPE_RSA:
+    keytype = CURLKHTYPE_RSA;
+    break;
+  case LIBSSH2_HOSTKEY_TYPE_DSS:
+    keytype = CURLKHTYPE_DSS;
+    break;
+  case LIBSSH2_HOSTKEY_TYPE_ECDSA_256:
```

Comment:
If I see correctly, LIBSSH2_HOSTKEY_TYPE_ECDSA_* and LIBSSH2_HOSTKEY_TYPE_ED25519 flags have been added in libssh2 library since 1.9.0 version. You need to add guards to watch over it. For example:

```suggestion
#ifdef LIBSSH2_HOSTKEY_TYPE_ECDSA_256
  case LIBSSH2_HOSTKEY_TYPE_ECDSA_256:
           /*FALLTHROUGH*/
#endif
```